### ISSUE TITLE
Change security slack notifications to alert based on open code scanning results

### ISSUE THREAD

**Issue by benmoss** [carvel-dev/kapp-controller#493]
Right now our [Trivy code scanning workflow](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/cbca53c348c20b7a95ff3a581a2d4cc8c146a4e8/.github/workflows/trivy-scan.yml) will send Slack alerts based on whether the Trivy CLI finds any unresolved issues in our images/binaries. It also posts them to GitHub's [Code Scanning](https://github.com/vmware-tanzu/carvel-kapp-controller/security/code-scanning) tool for our repo, which allows us to mark them as resolved/wontfix.

The problem is that if we have a wontfix it will be closed from GitHub, but Trivy will still report it as present. If we could trigger based on the open Code Scanning tickets instead we could have a better picture of unresolved items. It looks like it should be possible using [GitHub's API](https://docs.github.com/en/rest/reference/code-scanning).

If we get this working we should also fix this with other Carvel projects.

**Comment by benmoss** [comment]
This turned out to be slightly more complicated than I thought since we're scanning the latest release as well. I don't think there's a good way to associate a code scan alert with a release, so for instance if we had a fixed vulnerability in `develop` but that was still vulnerable in the latest release, I'm not sure it's simple to distinguish where it's fixed

**Comment by benmoss** [comment]
Reopening since this is gonna bother me forever. I looked into it a bit today but was blocked by https://github.com/github/codeql-action/issues/890. There looks like there will be a fix there shortly so I'll revisit once that's merged.
